omnigibson/prims/entity_prim.py

```diff
@@ -994,86 +995,110 @@ def get_relative_linear_velocity(self):
+    ):
+        """
+        Set the position and orientation of entry prim object.
+
+        Args:
+            position (None or 3-array): The position to set the object to. If None, the position is not changed.
+            orientation (None or 4-array): The orientation to set the object to. If None, the orientation is not changed.
+            frame (Literal): The frame in which to set the position and orientation. Defaults to world. parent frame
+            set position relative to the object parent. scene frame set position relative to the scene.
+        """
+
+        assert frame in ["world", "parent", "scene"], f"Invalid frame '{frame}'. Must be 'world', 'parent', or 'scene'."
 
-    def set_position_orientation(self, position=None, orientation=None):
-        position = (
-            position
-            if isinstance(position, th.Tensor)
-            else th.tensor(position, dtype=th.float32) if position is not None else None
-        )
-        orientation = (
-            orientation
-            if isinstance(orientation, th.Tensor)
-            else th.tensor(orientation, dtype=th.float32) if orientation is not None else None
-        )
         # If kinematic only, clear cache for the root link
         if self.kinematic_only:
             self.root_link.clear_kinematic_only_cache()
         # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
         if og.sim.is_stopped():
-            XFormPrim.set_position_orientation(self, position=position, orientation=orientation)
+            XFormPrim.set_position_orientation(self, position=position, orientation=orientation, frame=frame)
         # Delegate to RigidPrim if we are not articulated
         elif self._articulation_view is None:
-            self.root_link.set_position_orientation(position=position, orientation=orientation)
+            self.root_link.set_position_orientation(position=position, orientation=orientation, frame=frame)
         # Sim is running and articulation view exists, so use that physx API backend
         else:
-            if position is not None:
-                position = th.asarray(position)[None, :]
-            if orientation is not None:
-                orientation = th.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_world_poses(position, orientation)
+
+            # compute desired pose in the specified frame
+            position, orientation = T.compute_desired_pose_in_frame(self, position, orientation, frame=frame)
+
+            # convert to format expected by articulation view
+            position = th.asarray(position)[None, :]
+            orientation = th.asarray(orientation)[None, [3, 0, 1, 2]]
+
+            if frame == "world" or frame == "scene":
+                self._articulation_view.set_world_poses(position, orientation)
+            else:
+                self._articulation_view.set_local_poses(position, orientation)
+
```

Comment:
Restructure this part of the code to be a VERY CLEAR if else system:

```
if frame == "world":
   self._articulation_view.set_world_poses(position[None, :], orientation[None, [3, 0, 1, 2])
elif frame == "scene":
   position, orientation = mat2pose(prim.scene.pose @ pose2mat((position, orientation)))
   self._articulation_view.set_world_poses(position[None, :], orientation[None, [3, 0, 1, 2])
else:
   self._articulation_view.set_local_poses(position[None, :], orientation[None, [3, 0, 1, 2])
```

Notice how that is very clear in terms of EXACTLY what happens for EACH of the blocks. Currently this is confusing af - sometimes the utility function does things and sometimes it doesnt, then we have a if block that works with multiple of the options, etc. - lets clean this up.